### PR TITLE
Fix for Ash interaction with Top Hat and Khusyuk

### PR DESCRIPTION
**Pull request by lostgeek**

### CODE REVIEW

File: src/clj/game/cards/events.clj [hunk 1]
```diff
@@ -1328,11 +1328,12 @@
                                                                    "no cards")))
                                   (if revealed
                                     (do (reveal state side revealed)
-                                        (wait-for
-                                          (resolve-ability state side (access-revealed revealed) card nil)
-                                          (shuffle! state :corp :deck)
-                                          (system-msg state :runner " shuffles R&D")
-                                          (effect-completed state side eid)))
+                                        (if (not= (:max-access run) 0)
```
Comment (NoahTheDuke) [inline src/clj/game/cards/events.clj]:
```suggestion
                                        (when (not= (:max-access run) 0)
```

File: src/clj/game/cards/events.clj [hunk 2]
```diff
@@ -1328,11 +1328,12 @@
...
+                                          (do (wait-for
```
Comment (NoahTheDuke) [inline src/clj/game/cards/events.clj]:
```suggestion
                                          (wait-for
```
Don't actually commit this, just remove the `do` and then re-align everything.

File: test/clj/game_test/cards/events.clj [hunk 3]
```diff
@@ -2072,7 +2072,59 @@
...
+        (core/gain state :runner :credit 100)
+        (dotimes [_ 3] (play-from-hand state :runner "Cache"))
+        (play-run-event state (find-card "Khusyuk" (:hand (get-runner))) :rd)
+        (click-prompt state :corp "0")
+        (click-prompt state :runner "0") ;lose Ash trace
+        (click-prompt state :runner "Replacement effect")
+        (click-prompt state :runner "1 [Credit]: 3 cards")
+        (is (last-log-contains? state "Accelerated Beta Test, Brainstorm, Chiyashi") "Revealed correct 3 cards from R&D")
+        (is (empty? (:prompt (get-runner))) "No prompt to access cards.")
+        )
```
Comment (NoahTheDuke) [inline test/clj/game_test/cards/events.clj]: :eyes:

**Comment by lostgeek** [comment]
Latest commit closes #3779

**Pull request merged**